sdk/schemaregistry/Azure.Data.SchemaRegistry/tests/SchemaRegistryClientLiveTests.cs

```diff
@@ -25,38 +25,63 @@ public SchemaRegistryClientLiveTests(bool isAsync) : base(isAsync)
 
         private const string SchemaContent = "{\"type\" : \"record\",\"namespace\" : \"TestSchema\",\"name\" : \"Employee\",\"fields\" : [{ \"name\" : \"Name\" , \"type\" : \"string\" },{ \"name\" : \"Age\", \"type\" : \"int\" }]}";
 
-        [Test]
+        [RecordedTest]
         public async Task CanRegisterSchema()
         {
             var client = CreateClient();
             var schemaName = "test1";
             var groupName = TestEnvironment.SchemaRegistryGroup;
             var schemaType = SerializationType.Avro;
 
-            var schemaProperties = await client.RegisterSchemaAsync(groupName, schemaName, schemaType, SchemaContent);
-            Assert.IsNotNull(schemaProperties.Value);
-            Assert.IsNotNull(schemaProperties.Value.Id);
-            Assert.IsTrue(Guid.TryParse(schemaProperties.Value.Id, out Guid _));
-            Assert.AreEqual(SchemaContent, schemaProperties.Value.Content);
+            SchemaProperties registerProperties = await client.RegisterSchemaAsync(groupName, schemaName, schemaType, SchemaContent);
+            AssertSchemaProperties(registerProperties);
+
+            // this should be a cached lookup
+            var schemaProperties = await client.GetSchemaIdAsync(groupName, schemaName, schemaType, SchemaContent);
```

Comment:
is there a way to check whether a request was sent? in JS we have a callback that gets triggered only one received responses and I wonder if there is anything like that in .NET.

Reply:
I validated that the test recordings have the expected requests. If this ever broke, the tests would start failing playback as they would not match the existing recordings.

Reply:
To answer your question though, we could use a custom pipeline policy to do something like that (it would only be invoked if a request was being sent), but I think the test recording is sufficient.

Reply:
That makes sense. The risk is perhaps if caching breaks somehow right before you re-record the tests but I reckon the chances this happens are slim?

Reply:
Well you would have to validate the recordings again if re-recording. If I find that I need to re-record these a lot, I will add the custom policy 😄